design/cdk-bootstrap.md

```diff
@@ -0,0 +1,444 @@
+                "",
+                {
+                  "Ref": "TrustedAccounts"
+                }
+              ]
+            }
+          ]
+        }
+      ]
+    }
+  },
+  "Resources": {
+    "StagingBucket": {
+      "Type": "AWS::S3::Bucket",
+      "Properties": {
+        "BucketName": {
+          "Fn::Sub": "${ResourceNamePrefix}-${StagingBucketName}-${AWS::AccountId}-${AWS::Region}"
+        },
+        "AccessControl": "Private",
+        "BucketEncryption": {
+          "ServerSideEncryptionConfiguration": [{
+            "ServerSideEncryptionByDefault": {
+              "SSEAlgorithm": "aws:kms",
+              "KMSMasterKeyID": {
+                "Fn::If": [
+                  "HasStagingBucketKmsKeyId",
+                  { "Ref":  "StagingBucketKmsKeyId" },
+                  { "Ref":  "AWS::NoValue" }
+                ]
+              }
+            }
+          }]
+        },
+        "PublicAccessBlockConfiguration": {
+          "BlockPublicAcls": true,
+          "BlockPublicPolicy": true,
+          "IgnorePublicAcls": true,
+          "RestrictPublicBuckets": true
+        }
+      }
+    },
+    "DockerAssetsStagingRepository": {
+      "Type": "AWS::ECR::Repository",
+      "Properties": {
+        "RepositoryName": {
+          "Fn::Sub": "${ResourceNamePrefix}-${DockerAssetsStagingRepositoryName}-${AWS::AccountId}-${AWS::Region}"
+        }
+      }
+    },
+    "PublishingRole": {
+      "Type": "AWS::IAM::Role",
+      "Properties": {
+        "AssumeRolePolicyDocument": {
+          "Statement": [
+            {
+              "Action": "sts:AssumeRole",
+              "Effect": "Allow",
+              "Principal": {
+                "AWS": {
+                  "Ref": "AWS::AccountId"
+                }
+              }
+            }
+          ]
+        },
+        "Policies": [
+          {
+            "PolicyDocument": {
+              "Statement": [
+                {
+                  "Action": [
+                    "s3:GetObject*",    "s3:GetBucket*", "s3:List*",
+                    "s3:DeleteObject*", "s3:PutObject*", "s3:Abort*"
+                  ],
+                  "Resource": [
+                    {
+                      "Fn::Sub": "${StagingBucket.Arn}"
+                    },
+                    {
+                      "Fn::Sub": "${StagingBucket.Arn}/*"
+                    }
+                  ]
+                },
+                {
+                  "Action": [
+                    "ecr:PutImage",        "ecr:InitiateLayerUpload",
+                    "ecr:UploadLayerPart", "ecr:CompleteLayerUpload"
+                  ],
+                  "Resource": {
+                    "Fn::Sub": "${DockerAssetsStagingRepository.Arn}"
+                  }
+                }
+              ]
+            },
+            "PolicyName": {
+              "Fn::Sub": "DefaultBootstrapPublishRolePolicy-${AWS::Region}"
+            }
+          }
+        ],
+        "RoleName": "some-name"
```

Comment:
“Default”

Reply:
This needs to be a physical name based on the convention (prefix-xxx-account-region)

Reply:
Oops, I missed this one. Done.

Reply:
Actually, a question.

I understand why we need `account` in the bucket name (unique name, etc.). Are we also adding `account` to things that are not globally unique, like IAM roles?

Reply:
It feels like the right approach to use a common convention for all resources (both account and region)